src/datasources/errors/http-error-factory.ts

```diff
@@ -36,9 +37,8 @@ export class HttpErrorFactory {
   }
 }
 
-function isNetworkResponseError(
-  error: NetworkError,
-): error is NetworkResponseError {
-  const e = error as NetworkResponseError;
-  return e.data !== undefined && e.status !== undefined;
+function isNetworkResponseError(error: any): error is NetworkResponseError {
+  return (
+    error.status !== undefined && error.status >= 400 && error.status < 600
+  );
```

Comment:
I think the `!== undefined` check should be covered by the error range check, right? I mean, if `status` is in range it is defined, and if is undefined, it isn't in range. Maybe I'm missing something... 

Apart from that, why is `any` used here? I mean, `unknown` could be used, or even `error: NetworkError | Error)`, and keep the cast.

```javascript
function isNetworkResponseError(error: NetworkError | Error) ...
```

Reply:
I agree. Updated!